src/raftstore/store/worker/read.rs

```diff
@@ -0,0 +1,760 @@
+
+use rocksdb::DB;
+use raftstore::Result;
+use raftstore::store::{check_epoch, exec_read, Msg as StoreMsg, ReadResponse, RequestPolicy};
+use raftstore::store::util::{LeaseState, RemoteLease};
+use raftstore::errors::RAFTSTORE_IS_BUSY;
+use util::worker::Runnable;
+use util::transport::{NotifyError, Sender};
+use util::transport::RetryableSendCh;
+use util::collections::{HashMap, HashMapEntry};
+
+use super::metrics::*;
+
+/// Status for leaders
+#[derive(Debug)]
+pub struct LeaderStatus {
+    region: Region,
+    leader: Peer,
+    term: u64,
+    applied_index_term: u64,
+    leader_lease: Option<Arc<RemoteLease>>,
+
+    tag: String,
+}
+
+impl LeaderStatus {
+    fn merge(&mut self, status: LeaderStatus) {
+        self.region = status.region;
+        self.leader = status.leader;
+        self.tag = status.tag;
+
+        if self.term <= status.term {
+            self.term = status.term;
+        } else {
+            warn!(
+                "Stale LeaderStatus, cached term {}, update term {}",
+                self.term, status.term
+            );
+        }
+        if self.applied_index_term <= status.applied_index_term {
+            self.applied_index_term = status.applied_index_term;
+        } else {
+            warn!(
+                "Stale LeaderStatus, cached applied_index_term {}, update applied_index_term {}",
+                self.applied_index_term, status.applied_index_term
+            );
+        }
+        if let Some(lease) = status.leader_lease {
+            self.leader_lease = Some(lease);
+        }
+    }
+}
+
+impl Display for LeaderStatus {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        write!(
+            f,
+            "LeaderStatus for region {}, \
+             leader {} at term {}, applied_index_term {}, has lease {}",
+            self.region.get_id(),
+            self.leader.get_id(),
+            self.term,
+            self.applied_index_term,
+            self.leader_lease.is_some(),
+        )
+    }
+}
+
+pub enum Task {
+    Msg(StoreMsg),
+    Update(LeaderStatus),
+    Invalidate(u64),
+}
+
+impl Task {
+    pub fn update(
+        region: Region,
+        leader: Peer,
+        term: u64,
+        applied_index_term: u64,
+        leader_lease: Option<Arc<RemoteLease>>,
+    ) -> Task {
+        let tag = format!("[region {}]", region.get_id());
+        Task::Update(LeaderStatus {
+            region,
+            leader,
+            term,
+            applied_index_term,
+            tag,
+            leader_lease,
+        })
+    }
+
+    pub fn invalidate(region_id: u64) -> Task {
+        Task::Invalidate(region_id)
+    }
+
+    #[inline]
+    pub fn acceptable(msg: &StoreMsg) -> bool {
+        match *msg {
+            StoreMsg::RaftCmd { ref request, .. } => {
+                if request.has_admin_request() || request.has_status_request() {
+                    false
+                } else {
+                    for r in request.get_requests() {
```

Comment:
Why not write `request.get_requests().iter().all(|&r| r.get_cmd_type() == CmdType::Get)`?

Reply:
ping here

Reply:
I think a `for` loop is more readable.